### ISSUE TITLE
Fails to make on MacOS 10.14.16 with Xcode 11.3

### ISSUE THREAD

**Issue by quentinnuk** [mbarbon/extutils-cppguess#1]
```
[... snip ...]
t/001_load.t ........... ok
t/002_icpp.t ........... ok
error: invalid value 'c++unreal' in '-std=c++unreal'
note: use 'c++98' or 'c++03' for 'ISO C++ 1998 with amendments' standard
note: use 'gnu++98' or 'gnu++03' for 'ISO C++ 1998 with amendments and GNU extensions' standard
note: use 'c++11' for 'ISO C++ 2011 with amendments' standard
note: use 'gnu++11' for 'ISO C++ 2011 with amendments and GNU extensions' standard
note: use 'c++14' for 'ISO C++ 2014 with amendments' standard
note: use 'gnu++14' for 'ISO C++ 2014 with amendments and GNU extensions' standard
note: use 'c++17' for 'ISO C++ 2017 with amendments' standard
note: use 'gnu++17' for 'ISO C++ 2017 with amendments and GNU extensions' standard
note: use 'c++2a' for 'Working draft for ISO C++ 2020' standard
note: use 'gnu++2a' for 'Working draft for ISO C++ 2020 with GNU extensions' standard
t/003_standard_flag.t .. ok
# Module::Build version: 0.4229
# ExtUtils::CBuilder version: 0.280236

#   Failed test 'build with Module::Build'
#   at t/010_module_build.t line 19.
# Build.PL output
# ========================================
# clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
# Created MYMETA.yml and MYMETA.json
# Creating new 'Build' script for 'CppGuessTest' version '0.01'
# ========================================
# Build output
# ========================================
# Building CppGuessTest
# clang++ -I/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE -DVERSION="0.01" -DXS_VERSION="0.01" -xc++ -DINCLUDE_DOT=0 -Wno-reserved-user-defined-literal -c -g -pipe -fno-common -DPERL_DARWIN -fno-strict-aliasing -fstack-protector -Wno-error=implicit-function-declaration -Os -o lib/CppGuessTest.o lib/CppGuessTest.c
# lib/CppGuessTest.xs:18:10: fatal error: 'config.h' file not found
# #include <config.h>
#          ^~~~~~~~~~
# 1 error generated.
# error building lib/CppGuessTest.o from 'lib/CppGuessTest.c' at /Library/Perl/5.18/ExtUtils/CBuilder/Base.pm line 185.
# ========================================
# Looks like you failed 1 test of 1.
t/010_module_build.t ... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
t/011_makemaker.t ...... ok

Test Summary Report
-------------------
t/010_module_build.t (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=6, Tests=17,  8 wallclock secs ( 0.03 usr  0.01 sys +  2.75 cusr  1.08 csys =  3.87 CPU)
Result: FAIL
Failed 1/6 test programs. 1/17 subtests failed.
make: *** [test_dynamic] Error 255
-> FAIL Installing ExtUtils::CppGuess failed. See /Users/administator/.cpanm/work/1625608820.977/build.log for details. Retry with --force to force install it.
-> FAIL Installing the dependencies failed: Module 'ExtUtils::CppGuess' is not installed
-> FAIL Bailing out the installation for re-engine-RE2-0.14.
```

```
$ clang --version
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

$ perl -V
Summary of my perl5 (revision 5 version 18 subversion 4) configuration:
   
  Platform:
    osname=darwin, osvers=18.0, archname=darwin-thread-multi-2level
    uname='darwin osx219.apple.com 18.0 darwin kernel version 17.0.0: fri may 4 10:33:38 pdt 2018; root:xnu-4570.1.46.100.2~1development_x86_64 x86_64 '
    config_args='-ds -e -Dprefix=/usr -Dccflags=-g  -pipe  -Dldflags= -Dman3ext=3pm -Duseithreads -Duseshrplib -Dinc_version_list=none -Dcc=cc'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags =' -g -pipe -fno-common -DPERL_DARWIN -fno-strict-aliasing -fstack-protector',
    optimize='-Os',
    cppflags='-g -pipe -fno-common -DPERL_DARWIN -fno-strict-aliasing -fstack-protector'
    ccversion='', gccversion='4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.37.14)', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector'
    libpth=/usr/lib /usr/local/lib
    libs= 
    perllibs=
    libc=, so=dylib, useshrplib=true, libperl=libperl.dylib
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=bundle, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -bundle -undefined dynamic_lookup -fstack-protector'


Characteristics of this binary (from libperl): 
  Compile-time options: HAS_TIMES MULTIPLICITY PERLIO_LAYERS
                        PERL_DONT_CREATE_GVSV
                        PERL_HASH_FUNC_ONE_AT_A_TIME_HARD
                        PERL_IMPLICIT_CONTEXT PERL_MALLOC_WRAP
                        PERL_PRESERVE_IVUV PERL_SAWAMPERSAND USE_64_BIT_ALL
                        USE_64_BIT_INT USE_ITHREADS USE_LARGE_FILES
                        USE_LOCALE USE_LOCALE_COLLATE USE_LOCALE_CTYPE
                        USE_LOCALE_NUMERIC USE_PERLIO USE_PERL_ATOF
                        USE_REENTRANT_API
  Locally applied patches:
	/Library/Perl/Updates/<version> comes before system perl directories
	installprivlib and installarchlib points to the Updates directory
  Built under darwin
  Compiled at Oct 26 2020 19:51:48
  @INC:
    /Library/Perl/5.18/darwin-thread-multi-2level
    /Library/Perl/5.18
    /Network/Library/Perl/5.18/darwin-thread-multi-2level
    /Network/Library/Perl/5.18
    /Library/Perl/Updates/5.18.4/darwin-thread-multi-2level
    /Library/Perl/Updates/5.18.4
    /System/Library/Perl/5.18/darwin-thread-multi-2level
    /System/Library/Perl/5.18
    /System/Library/Perl/Extras/5.18/darwin-thread-multi-2level
    /System/Library/Perl/Extras/5.18

```

**Comment by zmughal** [comment]
The current repo and issue tracker is at <https://github.com/tsee/extutils-cppguess>.

Are you still having this issue? It looks like this is using the Perl that comes with macOS. That's not recommended as that Perl is set up for Apple to use for their own tools which they can upgrade at any time (it is recommended to use something like `perlbrew` or a package manager instead).

The error seems indicate that the system is missing the basic headers that come with Perl that would be under the include path `-I/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE` which means the install does not have what is needed to compile XS code.

**Comment by quentinnuk** [comment]
This is so old I think its probably been addressed and my perl version is now 5.36 and is no longer system perl, so I will close it and if it ever comes up again will relog.
